server/athenian/api/__main__.py

```diff
@@ -298,12 +300,15 @@ def create_segment() -> Optional[SegmentClient]:
 PRELOADER_VAR_NAME = "mc_preloader"
 
 
-def setup_preloading(app: AthenianApp, log: logging.Logger) -> None:
+def setup_preloading(app: AthenianApp, preload_refresh_frequency: int,
+                     log: logging.Logger) -> None:
     """Initialize the memory cache and schedule loading the DB tables."""
     log.info("Preloading DB tables to memory is enabled")
     app.app[PRELOADER_VAR_NAME] = mc_preloader = MemoryCachePreloader(
-        app.app[PROMETHEUS_REGISTRY_VAR_NAME])
+        preload_refresh_frequency,
+        prometheus_registry=app.app[PROMETHEUS_REGISTRY_VAR_NAME])
     app.on_dbs_connected(mc_preloader.preload)
+    app.on_shutdown(mc_preloader.stop)
```

Comment:
```suggestion
    app.app.on_shutdown.append(mc_preloader.stop)
```
and we don't need our custom code in `connexion.py` :+1:

Reply:
The reason why I did it is to have all the shutdown operations centralized in the current `_shutdown` method. In this case, for example, these callbacks will be called after the check of `self._shutting_down` and after calling `self._set_unready()`.

Reply:
Too much code for a single use-case. Let's iterate over `self.app.on_shutdown` inside our `shutdown()` then and remove `self.app.on_shutdown.append(self.shutdown)`.

Reply:
And how our `shutdown()` is triggered then if we remove `self.app.on_shutdown.append(self.shutdown)`?

Reply:
The thing is, we actually disable the built-in shutdown mechanism, the signal handlers, etc. because they do not work good enough: https://github.com/athenianco/athenian-api/blob/master/server/athenian/api/connexion.py#L648
I think I inserted `self.app.on_shutdown.append(self.shutdown)` during a brain freeze.

Reply:
I think there's something that I'm clearly missing. Even if you remove and add the signal handlers, I don't see any other place for calling `self.shutdown` other than being appended into `self.app.on_shutdown`.

Reply:
In short, we intercept SIGTERM and SIGINT and do not let aiohttp do the standard job. We further re-raise GracefulExit if needed. It works quite twisted, but super reliable.

Reply:
Now I see. Yes, GracefulExit in turn triggers the pass over `app.on_shutdown`, so we shouldn't remove it. OK! Then the only thing to do is `self.app.on_shutdown.insert(0, callback)` and the rest will happen automatically.

What I initially thought is that we overrode the `shutdown()` that's called before the signal processing. No, it's the other way round.

Reply:
done